src/database/utilities/createRecord.js

```diff
@@ -661,10 +661,14 @@ const createItemBatch = (database, item, batchString, supplier) => {
       .filtered('itemId == $0 && joinsThisStore == true', itemId)[0];
     const { restrictedLocationType } = itemStoreJoin ?? {};
     const { id: locationTypeId = '' } = restrictedLocationType ?? {};
-    const location = database.objects('Location').filtered('locationType.id == $0', locationTypeId);
-
-    itemBatch.applyLocation(database, location);
-    itemBatch.applyVvmStatus(database, vaccineVialMonitorStatus);
+    const location = database
+      .objects('Location')
+      .filtered('locationType.id == $0', locationTypeId)[0];
+
+    if (itemBatch.shouldApplyLocation(location)) itemBatch.applyLocation(database, location);
+    if (itemBatch.shouldApplyVvmStatus(vaccineVialMonitorStatus)) {
+      itemBatch.applyVvmStatus(database, vaccineVialMonitorStatus);
```

Comment:
added these checks to check if it should apply or not - shouldv'e been added originally, oops